debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoUtil.java

```diff
@@ -341,7 +341,8 @@ protected static ServerAddress getPrimaryAddress(MongoClient client) {
         if (serverDescriptions == null || serverDescriptions.size() == 0) {
             throw new DebeziumException("Unable to read server descriptions from MongoDB connection, got '" + serverDescriptions + "'");
         }
-        return new ServerAddress(serverDescriptions.get(0).getPrimary());
+        ServerAddress serverAddress = serverDescriptions.get(0).getAddress();
+        return new ServerAddress(serverAddress.getHost(), serverAddress.getPort());
```

Comment:
When omitting the port, the default port will be used, which is wrong when Mongo DB runs on a different port.